kong/clustering/control_plane.lua

```diff
@@ -357,30 +357,31 @@ function _M:handle_cp_websocket(cert)
       if typ == "close" then
         ngx_log(ngx_DEBUG, _log_prefix, "received close frame from data plane", log_suffix)
         return
-      end
 
-      if not data then
-        return nil, "did not receive ping frame from data plane"
+      elseif typ == "ping" then
+        ngx_log(ngx_DEBUG, _log_prefix, "received ping frame from data plane", log_suffix)
 
-      elseif #data ~= 32 then
-        return nil, "received a ping frame from the data plane with an invalid"
-                 .. " hash: '" .. tostring(data) .. "'"
-      end
+        if not data then
+          return nil, "did not receive ping frame from data plane"
 
-      -- dps only send pings
-      if typ ~= "ping" then
-        return nil, "invalid websocket frame received from data plane: " .. typ
-      end
+        elseif #data ~= 32 then
+          return nil, "received a ping frame from the data plane with an invalid"
+                   .. " hash: '" .. tostring(data) .. "'"
+        end
 
-      ngx_log(ngx_DEBUG, _log_prefix, "received ping frame from data plane", log_suffix)
+        config_hash = data
+        last_seen = ngx_time()
+        update_sync_status()
 
-      config_hash = data
-      last_seen = ngx_time()
-      update_sync_status()
+        -- queue PONG to avoid races
+        table_insert(queue, PONG_TYPE)
+        queue.post()
 
-      -- queue PONG to avoid races
-      table_insert(queue, PONG_TYPE)
-      queue.post()
+      -- dps only send pings
```

Comment:
Minor comment:
Is this comment outdated now?

I suggest explicitly handling the binary message to keep the protocol cleaner. Something like:
if the CP gets a binary frame, check the type, if it is 'error' then print in logs along the lines of "DP reports back errors in configuration". If you get any other type of message in the binary frame, "protocol error" in the logs

This explicit handling makes the CP aware of the new message type and the behavior of CP around handling the error can be improved in subsequent versions.

Reply:
Good feedback, and I agree. Once there is more consensus about removing the Konnect exclusivity from the feature I will make the Lua CP code a little more robust.